server/src/main/java/org/opensearch/env/NodeEnvironment.java

```diff
@@ -1143,6 +1214,22 @@ private void ensureNoShardData(final NodePath[] nodePaths) throws IOException {
         }
     }
 
+    /**
+     * Throws an exception if cache exists on a non-search node.
+     */
+    private void ensureNoCacheData(final NodePath cacheNodePath) throws IOException {
```

Comment:
I think we should have a `FCMonitorService` to keep track of the cache stats. By design, stat calls do not logically belong within the `NodeEnvironment` instance.